.travis.yml

```diff
@@ -1,24 +1,19 @@
-dist: xenial
-language: python
-sudo: true
-branches:
-  only:
-    - master
-
-python:
-  - 3.6
-  - 3.7
+language: minimal
+sudo: false
 
 env:
   - PYSAL_PYPI=true MPLBACKEND='pdf'
   - PYSAL_PYPI=false MPLBACKEND='pdf'
 
 matrix:
+  fast_finish: true
   allow_failures: #allow travis tests to fail if using the github version of libpysal
-      - python: 3.6
-        env: PYSAL_PYPI=false MPLBACKEND='pdf'
-      - python: 3.7
-        env: PYSAL_PYPI=false MPLBACKEND='pdf'
+      - name: "python-3.6"
```

Comment:
**Note:**
 Should each version of Python in the matrix have a `PYSAL_PYPI=true` and a `PYSAL_PYPI=false`?

Reply:
**Note:**
How to accommodate `allow_failures` and `include` to perform all combinations in testing matrix?